src/test/java/org/dependencytrack/integrations/defectdojo/DefectDojoClientTest.java

```diff
@@ -88,31 +85,51 @@ public void testUploadFindingsPositiveCase() throws Exception {
         DefectDojoUploader uploader = new DefectDojoUploader();
         DefectDojoClient client = new DefectDojoClient(uploader, new URL("https://localhost/defectdojo"));
         client.uploadDependencyTrackFindings(token, engagementId, new NullInputStream(0));
+        testClient.verify(
+                request()
+                .withMethod("POST")
+                .withPath("/defectdojo/api/v2/import-scan/"),
+                VerificationTimes.exactly(1)
+        );
+        testClient.clear(
+                request()
+                    .withPath("/defectdojo/api/v2/import-scan/")
+        );
     }
 
     @Test
     public void testUploadFindingsNegativeCase() throws Exception {
-        String token = "db975c97-98b1-4988-8d6a-9c3e044dfff3";
+        String token = "db975c97-98b1-4988-8d6a-9c3e044dfff2";
         String engagementId = "";
-        new MockServerClient("localhost", 1080)
-                .when(
+        MockServerClient testClient = new MockServerClient("localhost", 1080);
+        testClient.when(
                         request()
                                 .withMethod("POST")
                                 .withHeader(HttpHeaders.AUTHORIZATION, "Token " + token)
-                                .withPath("/defectdojo/api/v2/import-scan")
-                                .withBody(
-                                    params(
-                                        param("engagement", engagementId)
-                                    )
-                                )
+                                .withPath("/defectdojo/api/v2/import-scan/")
                 )
                 .respond(
                         response()
                                 .withStatusCode(400)
                                 .withHeader(HttpHeaders.CONTENT_TYPE, "application/json")
                 );
-        DefectDojoUploader uploader = new DefectDojoUploader();
+        DefectDojoUploader uploader = Mockito.mock(
+                DefectDojoUploader.class,
+                Mockito.CALLS_REAL_METHODS
+        );
         DefectDojoClient client = new DefectDojoClient(uploader, new URL("https://localhost/defectdojo"));
         client.uploadDependencyTrackFindings(token, engagementId, new NullInputStream(16));
+        testClient.verify(
+                request()
+                .withMethod("POST")
+                .withHeader(HttpHeaders.AUTHORIZATION, "Token " + token)
+                .withPath("/defectdojo/api/v2/import-scan/"),
+                VerificationTimes.exactly(1)
+        );
+        Mockito.verify(uploader, Mockito.times(1)).handleUnexpectedHttpResponse(Mockito.any(), Mockito.eq("https://localhost/defectdojo/api/v2/import-scan/"), Mockito.eq(400), Mockito.any());
```

Comment:
I'm a bit unclear as to the usage of Mockito in this Integration test as it feels kinda wrong to use a class mocking technique and a remote system mocking technique in the same class.  

I can see that you are trying to verify that a Bad Request is received but it feels like that is testing the MockServer framework more than the code.  

Perhaps the `testClient.verify()` invocation to prove that the API is called as expected and the fact that an Exception is not raised by the client is enough?

```
    @Test
    public void testUploadFindingsNegativeCase() throws Exception {
        String token = "db975c97-98b1-4988-8d6a-9c3e044dfff2";
        String engagementId = "";
        MockServerClient testClient = new MockServerClient("localhost", 1080);
        testClient.when(
                        request()
                                .withMethod("POST")
                                .withHeader(HttpHeaders.AUTHORIZATION, "Token " + token)
                                .withPath("/defectdojo/api/v2/import-scan/")
                )
                .respond(
                        response()
                                .withStatusCode(400)
                                .withHeader(HttpHeaders.CONTENT_TYPE, "application/json")
                );
        DefectDojoUploader uploader = new DefectDojoUploader();
        DefectDojoClient client = new DefectDojoClient(uploader, new URL("https://localhost/defectdojo"));
        try {
            client.uploadDependencyTrackFindings(token, engagementId, new NullInputStream(16));
        } catch (Exception ex) {
            Assert.fail("Exception not expected", ex);
        }
        testClient.verify(
                request()
                        .withMethod("POST")
                        .withHeader(HttpHeaders.AUTHORIZATION, "Token " + token)
                        .withPath("/defectdojo/api/v2/import-scan/"),
                VerificationTimes.exactly(1)
        );
        testClient.clear(
                request()
                        .withPath("/defectdojo/api/v2/import-scan/")
        );
    }
```

Reply:
Ok yeah - if you think that's enough of an assertion I'll change it back to the same assertion as the other test. I felt that without testing the error, it was basically the same as the positive test case.

Reply:
Yeah I think so, the client code doesn't handle exceptions but Unirest should be able to handle the 400 bad request without raising an exception.  It is essentially proving that the library is being used correctly and failing as expected.